src/integrations/Braze/browser.js

```diff
@@ -246,78 +180,72 @@ class Braze {
-        if (!prevTraits[key] || !isEqual(prevTraits[key], traits[key])) {
-          window.braze.getUser().setCustomUserAttribute(key, traits[key]);
-        }
-      });
+      if (birthday && !isEqual(birthday, prevBirthday)) setBirthday();
+      if (firstName && firstName !== prevFirstname) setFirstName();
+      if (lastName && lastName !== prevLastname) setLastName();
+      if (gender && formatGender(gender) !== formatGender(prevGender))
+        setGender(formatGender(gender));
+      if (address && !isEqual(address, prevAddress)) setAddress();
+      if (isObject(traits)) {
+        Object.keys(traits)
+          .filter((key) => reserved.indexOf(key) === -1)
+          .forEach((key) => {
+            if (!prevTraits[key] || !isEqual(prevTraits[key], traits[key])) {
+              window.braze.getUser().setCustomUserAttribute(key, traits[key]);
+            }
+          });
+      }
     } else {
       window.braze.changeUser(userId);
       // method removed from v4 https://www.braze.com/docs/api/objects_filters/user_attributes_object#braze-user-profile-fields
       // window.braze.getUser().setAvatarImageUrl(avatar);
       if (email) setEmail();
-      if (firstname) setFirstName();
-      if (gender) setGender(this.formatGender(gender));
-      if (lastname) setLastName();
+      if (firstName) setFirstName();
+      if (lastName) setLastName();
+      if (gender) setGender(formatGender(gender));
       if (phone) setPhone();
       if (address) setAddress();
       if (birthday) setBirthday();
-
-      Object.keys(traits).forEach((key) => {
-        window.braze.getUser().setCustomUserAttribute(key, traits[key]);
-      });
+      if (isObject(traits)) {
+        Object.keys(traits)
+          .filter((key) => reserved.indexOf(key) === -1)
+          .forEach((key) => {
+            window.braze.getUser().setCustomUserAttribute(key, traits[key]);
+          });
+      }
     }
-    this.previousPayload = { ...this.previousPayload, rudderElement };
-  }
-
-  handlePurchase(properties, userId) {
-    const { products } = properties;
-    const currencyCode = properties.currency;
-
-    window.braze.changeUser(userId);
-
-    // del used properties
-    del(properties, 'products');
-    del(properties, 'currency');
-
-    // we have to make a separate call to appboy for each product
-    if(products && Array.isArray(products) && products.length > 0) {
-        products.forEach((product) => {
-          const productId = product.product_id;
-          const { price } = product;
-          const { quantity } = product;
-          if (quantity && price && productId)
-            window.braze.logPurchase(productId, price, currencyCode, quantity, properties);
-        });
+    if (
+      this.supportDedup &&
+      isObject(previousPayload) &&
+      !R.isEmpty(previousPayload) &&
+      userId === previousPayload?.userId
+    ) {
+      Storage.setItem('rs_braze_dedup_attributes', { ...previousPayload, ...message });
```

Comment:
Yes that was considered, since its the same source running in same top domain which represents the organisation implementing the same source we will be good with this